File: 2580-circular-sentence/circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        if (sentence.front() != sentence.back()) {
            return false;
        }
        for (int i = 1; i < sentence.length() - 1; i++) {
            if (sentence[i] == ' ') {
                if (sentence[i - 1] != sentence[i + 1]) {
                    return false;
                }
            }
        }
        return true;
    }
};

File: 2580-circular-sentence/circular-sentence.js
var isCircularSentence = function (sentence) {
    if (sentence[0] !== sentence[sentence.length - 1]) {
        return false
    }
    for (let i = 1; i < sentence.length - 1; i++) {
        if (sentence[i] === " ") {
            if (sentence[i - 1] !== sentence[i + 1]) {
                return false
            }
        }
    }
    return true
};